arcgis-ios-sdk-samples/Edit data/Edit feature attachments/AttachmentsTableViewController.swift

```diff
@@ -0,0 +1,174 @@
+// Copyright 2016 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class AttachmentsTableViewController: UITableViewController {
+    
+    weak var feature: AGSArcGISFeature?
+    private var attachments: [AGSAttachment] = []
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        loadAttachments()
+    }
+    
+    private func loadAttachments() {
+        
+        //show progress hud
+        SVProgressHUD.show(withStatus: "Loading attachments")
+        
+        feature?.fetchAttachments { [weak self] (attachments: [AGSAttachment]?, error: Error?) in
+            
+            //dismiss progress hud
+            SVProgressHUD.dismiss()
+            
+            guard let self = self else {
+                return
+            }
+            
+            if let error = error {
+                self.presentAlert(error: error)
+            }
+            
+            self.attachments = attachments ?? []
```

Comment:
This might make a little more sense as an `else if`, since we know that either `attachments` or `error` will be non-`nil`. I think that's how we do it in most other places:

```swift
} else if let attachments = attachments {
    self.attachments = attachments
    self.tableView.reloadData()
}
```

Reply:
In this case we want the table to be cleared if there was an error so as not to show out-of-date data. I'll still change it to make that clearer.

Reply:
That makes sense. I wrote that comment before I realized that this is being called for additions and deletions.